### PR TITLE
ESD-1836: added additional massif options

### PR DESCRIPTION
**Pull request by RReichert**
Not much going on in here, except just adding some extra valgrind-massif options.

### CODE REVIEW

File: Valgrind.cmake [hunk 1]
```diff
@@ -109,6 +109,18 @@
 #
 ### MASSIF SPECIFIC OPTIONS:
 #
+# DEPTH maximum depth of the allocation trees recorded for detailed snapshots.
+#
```
Comment (Davinco) [inline Valgrind.cmake:113]: Could you also add a unit or range to use as argument to depth?

File: Valgrind.cmake [hunk 2]
```diff
@@ -109,6 +109,18 @@
...
+# DETAILED_FREQUENCY frequency of detailed snapshots. With value of `1`,
+# every snapshot is detailed.
+#
```
Comment (Davinco) [inline Valgrind.cmake:116]: Same comment as above, can I use decimal numbers as argument?

File: Valgrind.cmake (hunk 1 above)
Comment (RReichert) [reply Valgrind.cmake:115]: I did mention that the parameter expects a number and specified the default value. I've made this change to all the other single variable parameters as well.

There is no upper range within Valgrinds documentation (in my implementation it caps at 200), so I didn't want to specify the range as that isn't set in stone from the looks of things.

File: Valgrind.cmake (hunk 2 above)
Comment (RReichert) [reply Valgrind.cmake:118]: :heavy_check_mark:, marked it as `float`, the notation that is used in Valgrind of `m.n` seems a bit strange

Comment (RReichert) [reply Valgrind.cmake:118]: If you prefer `decimal` over `float` I can use `decimal`

Comment (Davinco) [reply Valgrind.cmake:118]: No, I prefer float :P looks good :+1:

File: Valgrind.cmake (hunk 1 above)
Comment (Davinco) [reply Valgrind.cmake:115]: Super! :+1:

**Pull request merged**